lib/data_layer.ex

```diff
@@ -1289,6 +1289,31 @@ defmodule AshPostgres.DataLayer do
       )
   end
 
+  defp upsert_fields(fields, _, _) when is_list(fields) do
+    fields
+  end
+
+  defp upsert_fields({:replace, fields}, _, _) do
+    fields
+  end
+
+  defp upsert_fields(:replace_all, resource, _) do
+    resource
+    |> Ash.Resource.Info.attributes()
+    |> Enum.map(fn %{name: name} -> name end)
+  end
+
+  defp upsert_fields({:replace_all_except, except_fields}, resource, _) do
+    resource
+    |> Ash.Resource.Info.attributes()
+    |> Enum.map(fn %{name: name} -> name end)
+    |> Enum.reject(fn name -> name in except_fields end)
+  end
+
+  defp upsert_fields(nil, _, changesets) do
+    changesets |> Enum.flat_map(&Map.keys(&1.attributes)) |> Enum.uniq()
+  end
```

Comment:
Not sure about this one. I wanted your opinion first.

After this is clear to me, I will also add some tests.

Reply:
This is interesting. This *should* be handled by the changeset logic before it gets here. I think we may need to fix this in core somewhere.

Reply:
Yeah, I think the `upsert_fields` option should be transformed to just be a list of fields by core before passing the option to the data layer.

Reply:
The problem is that it can also be provided after compile time as an option. So I think somewhere in bulk create logic we need to do the transformation. There should be similar logic in `Ash.Changeset`